io.go

```diff
@@ -6,16 +6,39 @@ import (
 	"io"
 )
 
-func Copy(ch chan<- []byte, r io.Reader) error {
+func Copy(ch chan<- []byte, r io.Reader, w io.Writer) error {
 	scanner := bufio.NewScanner(r)
 
 	for scanner.Scan() {
-		ch <- []byte(scanner.Text() + "\n")
+		line := scanner.Text() + "\n"
```

Comment:
I would probably use scanner.Bytes() here and stop with all of the conversions.  This does mean that you will need to pass a copy to `ch` though as the next call to Scan() could change the contents of the backing array. But that's as simple as:

```go
cpy = make([]byte, len(line))
copy(cpy, line)
ch <- cpy
```